### PR TITLE
[Gen3] Upgrades Nordic nRF5 SDK to 15.3.0

### PR DESCRIPTION
**Pull request by avtolstoy**
### References

- [CH32494]
- https://github.com/particle-iot/nrf5_sdk/tree/particle-15.3.0_59ac345

---

### Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [x] Problem and Solution clearly stated
- [x] Run unit/integration/application tests on device
- [x] N/A Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

---

- [enhancement] [gen3] Upgrades Nordic nRF5 SDK to 15.3.0 [#1768](https://github.com/particle-iot/device-os/pull/1768)

### CODE REVIEW

File: hal/src/argon/app_usbd_string_config.h [hunk 1]
```diff
@@ -85,7 +85,7 @@
     APP_USBD_STRING_DESC("Argon CDC Mode")
 #else // Argon SoM
 #define APP_USBD_STRINGS_PRODUCT         \
-    APP_USBD_STRING_DESC("ASoM CDC Mode")
+    APP_USBD_STRING_DESC("Argon SoM CDC Mode")
```
Comment (m-mcgowan) [inline hal/src/argon/app_usbd_string_config.h]: Why was this necessary? What other measures are needed before we can revert to the new naming?

Comment (avtolstoy) [reply hal/src/argon/app_usbd_string_config.h]: We have a separate story for this, which involves more than just USB strings and I'd like to keep all the changes isolated in a separate PR: [CH32184](https://app.clubhouse.io/particle/story/32184/rename-som-platform-names)

**Comment by avtolstoy** [comment]
@m-mcgowan AFAIK this is going into 1.2.1 not 1.2.0, right?

File: hal/src/argon/app_usbd_string_config.h (hunk 1 above)
Comment (technobly) [reply hal/src/argon/app_usbd_string_config.h]: It should be "A SoM CDC Mode" eventually.

Comment (avtolstoy) [reply hal/src/argon/app_usbd_string_config.h]: It's already resolved in this PR.

Comment (technobly) [reply hal/src/argon/app_usbd_string_config.h]: Sweet!  Didn't see this was an outdated comment.

**Comment by m-mcgowan** [comment]
> @m-mcgowan AFAIK this is going into 1.2.1 not 1.2.0, right?

Yep
